### PR TITLE
Adding delete roles and users PA-529

### PR DESCRIPTION
**Pull request by sn-px**
**What this PR does / why we need it**:
Validates the deletion of roles and users

creates 100 users , assigns roles 
un-assign roles and delete user

Logs : http://aetos.pwx.purestorage.com/resultSet/testSetID/81928/testCaseID/308694

**Which issue(s) this PR fixes** (optional)
PA-529
Closes #

logs after impl of comments: http://aetos.pwx.purestorage.com/resultSet/testSetID/85226/testCaseID/328275

### CODE REVIEW

**Review by mkoppal-px — CHANGES_REQUESTED** [review]
- The dashboard has some error in the end
- The delete role and user in the report does not tell me which user or which role it is deleting

File: tests/backup/backup_test.go [hunk 1]
```diff
@@ -4931,6 +4932,69 @@ var _ = Describe("{CustomResourceBackupAndRestore}", func() {
...
+					err = backup.DeleteUser(userName)
+					dash.VerifyFatal(err, nil, "Delete user")
+				}(userName,role)
+			}
+			wg.Wait()
+		})
+	})
+	JustAfterEach(func() {
+		defer EndTorpedoTest()
+		AfterEachTest(contexts, testrailID, runID)
```
Comment (mkoppal-px) [inline tests/backup/backup_test.go:4994]: Why is this needed?

**Comment by mkoppal-px** [comment]
After running this test, please login to keycloak and manually verify that the users are getting deleted

**Comment by mkoppal-px** [comment]
Add a validation step to get all users in the end and check that the created users are not present in it

File: tests/backup/backup_test.go (hunk 1 above)
Comment (sn-px) [reply tests/backup/backup_test.go:4994]: testrail push

**Comment by sn-px** [comment]
> * The dashboard has some error in the end
> * The delete role and user in the report does not tell me which user or which role it is deleting

done

**Comment by sn-px** [comment]
> After running this test, please login to keycloak and manually verify that the users are getting deleted

verified

**Comment by sn-px** [comment]
by mistake closed ..so reopening

**Comment by sn-px** [comment]
> Add a validation step to get all users in the end and check that the created users are not present in it

done

**Comment by sn-px** [comment]
> After running this test, please login to keycloak and manually verify that the users are getting deleted

done

**Comment by sn-px** [comment]
> After running this test, please login to keycloak and manually verify that the users are getting deleted

done

**Comment by sn-px** [comment]
> * The dashboard has some error in the end
> * The delete role and user in the report does not tell me which user or which role it is deleting

Done

**Comment by sn-px** [comment]
> * The dashboard has some error in the end
> * The delete role and user in the report does not tell me which user or which role it is deleting

Done

File: tests/backup/backup_test.go [hunk 2]
```diff
@@ -5708,6 +5711,80 @@ var _ = Describe("{ViewOnlyFullBackupRestoreIncrementalBackup}", func() {
 
 })
 
+var _ = Describe("{DeleteUsersRole}", func() {
+	var testrailID = 58089
```
Comment (mkoppal-px) [inline tests/backup/backup_test.go:5715]: No need to give this. Directly pass the ID in `StartTorpedoTest`

File: tests/backup/backup_test.go [hunk 3]
```diff
@@ -5708,6 +5711,80 @@ var _ = Describe("{ViewOnlyFullBackupRestoreIncrementalBackup}", func() {
...
+			}
+		})
+
+		Step("Validate if all the created users are deleted ", func(){
+			
+			var result bool = false
+
+			remainingUsers , _ := backup.GetAllUsers()
+
+			for user, _ := range userRoleMapping {
```
Comment (mkoppal-px) [inline tests/backup/backup_test.go:5771]: Remove the `_ ` here. It is not required.

File: tests/backup/backup_test.go [hunk 4]
```diff
@@ -5708,6 +5711,80 @@ var _ = Describe("{ViewOnlyFullBackupRestoreIncrementalBackup}", func() {
...
+					log.FailOnError(err, "Failed to add role to user - %s", userName)
+				}(userName, firstName, lastName, email, role)
+				userRoleMapping[userName] = role
+			}
+			wg.Wait()
+		})
+
+		Step("Delete roles and users", func() {
+
+			for userName,role := range userRoleMapping {
```
Comment (mkoppal-px) [inline tests/backup/backup_test.go:5756]: Add a space after the `,`

File: tests/backup/backup_test.go [hunk 5]
```diff
@@ -5708,6 +5711,80 @@ var _ = Describe("{ViewOnlyFullBackupRestoreIncrementalBackup}", func() {
...
+				log.Info("This is the user : ", userName)
+				err := backup.DeleteRoleFromUser(userName,role,"")
```
Comment (mkoppal-px) [inline tests/backup/backup_test.go:5758]: Add a spaces after the `,` everywhere

File: tests/backup/backup_test.go [hunk 6]
```diff
@@ -5708,6 +5711,80 @@ var _ = Describe("{ViewOnlyFullBackupRestoreIncrementalBackup}", func() {
...
+				dash.VerifyFatal(err, nil, fmt.Sprintf("Removing role [%s] from the user [%s]", role, userName))
+				err = backup.DeleteUser(userName)
+				dash.VerifyFatal(err, nil, fmt.Sprintf("Deleting the user [%s]", userName))
+			}
+		})
+
+		Step("Validate if all the created users are deleted ", func(){
```
Comment (mkoppal-px) [inline tests/backup/backup_test.go:5765]: Add space after `func()` before `{`

File: tests/backup/backup_test.go [hunk 7]
```diff
@@ -5708,6 +5711,80 @@ var _ = Describe("{ViewOnlyFullBackupRestoreIncrementalBackup}", func() {
...
+
```
Comment (mkoppal-px) [inline tests/backup/backup_test.go:5766]: Do not leave empty lines here

File: tests/backup/backup_test.go [hunk 8]
```diff
@@ -5708,6 +5711,80 @@ var _ = Describe("{ViewOnlyFullBackupRestoreIncrementalBackup}", func() {
...
+				err := backup.DeleteRoleFromUser(userName,role,"")
+				dash.VerifyFatal(err, nil, fmt.Sprintf("Removing role [%s] from the user [%s]", role, userName))
+				err = backup.DeleteUser(userName)
+				dash.VerifyFatal(err, nil, fmt.Sprintf("Deleting the user [%s]", userName))
+			}
+		})
+
+		Step("Validate if all the created users are deleted ", func(){
+			
+			var result bool = false
```
Comment (mkoppal-px) [inline tests/backup/backup_test.go:5767]: We can directly declare this like `bool := false`

File: tests/backup/backup_test.go [hunk 9]
```diff
@@ -5708,6 +5711,80 @@ var _ = Describe("{ViewOnlyFullBackupRestoreIncrementalBackup}", func() {
...
+
+			remainingUsers , _ := backup.GetAllUsers()
```
Comment (mkoppal-px) [inline tests/backup/backup_test.go:5769]: Remove the space after `remainingUsers`

**Review by mkoppal-px — CHANGES_REQUESTED** [review]
The test case should have steps in this order:

- Delete roles from users
- Validate that the roles are deleted
- Delete users
- Validate that the users are deleted

File: tests/backup/backup_test.go (hunk 2 above)
Comment (sn-px) [reply tests/backup/backup_test.go:5715]: declared since passed 2 places

File: tests/backup/backup_test.go (hunk 4 above)
Comment (sn-px) [reply tests/backup/backup_test.go:5756]: done

File: tests/backup/backup_test.go (hunk 9 above)
Comment (sn-px) [reply tests/backup/backup_test.go]: otherwise result gets scoped to for loop

File: tests/backup/backup_test.go (hunk 8 above)
Comment (sn-px) [reply tests/backup/backup_test.go:5761]: to avoid result getting scoped

File: tests/backup/backup_test.go (hunk 9 above)
Comment (sn-px) [reply tests/backup/backup_test.go]: done

File: tests/backup/backup_test.go (hunk 5 above)
Comment (sn-px) [reply tests/backup/backup_test.go:5754]: done

File: tests/backup/backup_test.go (hunk 1 above)
Comment (mkoppal-px) [reply tests/backup/backup_test.go:5776]: This is not needed. Testrail push happens in the beginning itself. After that it will just make APIs call to Aetos.

**Comment by sn-px** [comment]
> The test case should have steps in this order:
> 
> * Delete roles from users
> * Validate that the roles are deleted
> * Delete users
> * Validate that the users are deleted

done

File: tests/backup/backup_test.go [hunk 10]
```diff
@@ -5708,6 +5710,85 @@ var _ = Describe("{ViewOnlyFullBackupRestoreIncrementalBackup}", func() {
...
+		Step("Delete users", func() {
+			for userName, _ := range userRoleMapping {
+				log.Info("This is the user : ", userName)
+				err := backup.DeleteUser(userName)
+				dash.VerifyFatal(err, nil, fmt.Sprintf("Deleting the user [%s]", userName))
+			}
+		})
+		Step("Validate if all the created users are deleted ", func() {
+			var result bool = false
+			remainingUsers, _ := backup.GetAllUsers()
```
Comment (mkoppal-px) [inline tests/backup/backup_test.go:5775]: Handle the error returned here

File: tests/backup/backup_test.go (hunk 3 above)
Comment (mkoppal-px) [reply tests/backup/backup_test.go:5776]: I do not see this change

File: tests/backup/backup_test.go [hunk 11]
```diff
@@ -5708,6 +5710,85 @@ var _ = Describe("{ViewOnlyFullBackupRestoreIncrementalBackup}", func() {
...
+					if roleObj.Name == string(role) {
+						result = true
+						break
+					}	
+				}
+				dash.VerifyFatal(result, false, fmt.Sprintf("validation of deleted role [%s]", role))
+			}
+		})
+		Step("Delete users", func() {
+			for userName, _ := range userRoleMapping {
```
Comment (mkoppal-px) [inline tests/backup/backup_test.go:5767]: You can remove the underscore here

File: tests/backup/backup_test.go (hunk 3 above)
Comment (mkoppal-px) [reply tests/backup/backup_test.go:5776]: I do not see the change here

File: tests/backup/backup_test.go (hunk 5 above)
Comment (mkoppal-px) [reply tests/backup/backup_test.go:5749]: I do not see the change here

File: tests/backup/backup_test.go [hunk 12]
```diff
@@ -5708,6 +5710,85 @@ var _ = Describe("{ViewOnlyFullBackupRestoreIncrementalBackup}", func() {
...
+				log.Info("This is the user : ", userName)
+				err := backup.DeleteUser(userName)
+				dash.VerifyFatal(err, nil, fmt.Sprintf("Deleting the user [%s]", userName))
+			}
+		})
+		Step("Validate if all the created users are deleted ", func() {
```
Comment (mkoppal-px) [inline tests/backup/backup_test.go:5773]: Remove the space after `deleted`

File: tests/backup/backup_test.go (hunk 8 above)
Comment (mkoppal-px) [reply tests/backup/backup_test.go:5774]: I do not see the change here

**Review by mkoppal-px — CHANGES_REQUESTED** [review]
Please address all the comments and get another approval

File: tests/backup/backup_test.go (hunk 1 above)
Comment (sn-px) [reply tests/backup/backup_test.go]: removed

File: tests/backup/backup_test.go (hunk 3 above)
Comment (sn-px) [reply tests/backup/backup_test.go]: removed

File: tests/backup/backup_test.go (hunk 8 above)
Comment (sn-px) [reply tests/backup/backup_test.go]: done

File: tests/backup/backup_test.go (hunk 10 above)
Comment (sn-px) [reply tests/backup/backup_test.go]: done

File: tests/backup/backup_test.go (hunk 11 above)
Comment (sn-px) [reply tests/backup/backup_test.go]: done

File: tests/backup/backup_test.go (hunk 5 above)
Comment (sn-px) [reply tests/backup/backup_test.go]: done

**Comment by sn-px** [comment]
> Please address all the comments and get another approval

sure, looks like one of the push go missed , led to confusion ..have fixed the issue

**Review by snigdha-px — APPROVED** [review]
LGTM

**Review by lsrinivas-pure — APPROVED** [review]
LGTM

**Review by kshithijiyer-px — APPROVED** [review]
LGTM

**Review by mkoppal-px — APPROVED** [review]
LGTM
